src/Compilers/Core/MSBuildTask/ManagedToolTask.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.CodeAnalysis.BuildTasks
+{
+    public abstract class ManagedToolTask : ToolTask
+    {
+        protected abstract bool IsManagedTask { get; }
```

Comment:
I would call this `IsManagedTool` because Tasks are, by definition, always managed.